02-cpu-configure/cpu-configure.sh

```diff
@@ -2,108 +2,94 @@
-BPM_HALCS_CFG_FILE="/usr/local/etc/halcs/halcs.cfg"
-TIM_RX_EPICS_CFG_FILE="/etc/sysconfig/tim-rx-epics-ioc"
-
-## Ask sudo password only once and
-## keep updating sudo timestamp to
-## avoid asking again
-#sudo -v
-#while true; do sudo -n true; sleep 60; kill -0 "$$" || \
-#    exit; done 2>/dev/null &
-#
-## Install packages
-#sudo apt-get update && \
-#sudo apt-get install -y \
-#    openssh-client \
-#    sshpass
-
-
-exec_cmd "TRACE" echo "Generating BPM RFFE IP mapping file..."
-
-# Generate HALCS config from template
-BPM_MAX_NUM_BOARDS=12
-BPM_MAX_NUM_HALCS=2
-cp ${BPM_HALCS_CFG_TEMPLATE_IN_FILE} ${BPM_HALCS_CFG_TEMPLATE_OUT_FILE}
-for board in `seq 1 ${BPM_MAX_NUM_BOARDS}`; do
-    for halcs in `seq 0 $((${BPM_MAX_NUM_HALCS}-1))`; do
-        bpm_rffe_ip_idx="${board}${halcs}"
-        bpm_rffe_ip="BPM_RFFE_IP_MAPPING[${bpm_rffe_ip_idx}]"
-        bpm_rffe_proto="BPM_RFFE_PROTO_MAPPING[${bpm_rffe_ip_idx}]"
-        sed -i \
-            -e "s#<RFFE_BOARD${board}_HALCS${halcs}_IP>#${!bpm_rffe_ip}#" \
-            -e "s#<RFFE_BOARD${board}_HALCS${halcs}_PROTO>#${!bpm_rffe_proto}#" \
-            ${BPM_HALCS_CFG_TEMPLATE_OUT_FILE}
-    done
-done
-
-
-exec_cmd "TRACE" echo "Modifying EPICS PV prefixes..."
-
-# Login via SSH and setup configuration files
-# Set the --static, --transient and --pretty
-# hostnames by the HOSTNAME variable contents
-SSHPASS="${SSHPASS_USR}" sshpass -e \
-    ssh -o StrictHostKeyChecking=no \
-    root@${IP} \
-    bash -c "\
-        echo \"\" > /etc/hostname && \
-        sysctl kernel.hostname=\"${HOSTNAME}\" > /dev/null && \
-        sysctl -w kernel.hostname=\"${HOSTNAME}\" > /dev/null && \
-        hostnamectl set-hostname \"${HOSTNAME}\" > /dev/null && \
-        sed -i -e \"\
-            { \
-                s|EPICS_PV_CRATE_PREFIX=.*\$|EPICS_PV_CRATE_PREFIX=${CRATE}|; \
-            }\" ${BPM_EPICS_CFG_FILE} && \
-        sed -i -e \"\
-            { \
-                s|EPICS_PV_CRATE_PREFIX=.*\$|EPICS_PV_CRATE_PREFIX=${CRATE}|; \
-            }\" ${TIM_RX_EPICS_CFG_FILE} \
-    "
-
-exec_cmd "TRACE" echo "Copying generated files to CPU..."
-
-SSHPASS="${SSHPASS_USR}" sshpass -e \
-    scp \
-    ${BPM_HALCS_CFG_TEMPLATE_OUT_FILE} \
-    root@${IP}:${BPM_HALCS_CFG_FILE}
-
-exec_cmd "INFO " echo "CPU Configured Successfully!"
+hostnamectl set-hostname "$CRATE_FQDN"
+sysctl kernel.hostname="$CRATE_FQDN"
+sysctl -w kernel.hostname="$CRATE_FQDN"
+
+apt-get update -y
+apt-get upgrade -y
+apt-get dist-upgrade -y
+apt-get install build-essential vim tmux git rsync sshfs telnet screen picocom earlyoom procserv arch-install-scripts zabbix-agent2 tcpdump gdb strace wget lsb-release iptables iptables-persistent -y
+apt-get autoremove -y
+
+iptables -t nat -A PREROUTING -p udp ! -s 127.0.0.1/24 --dport 5064 -j DNAT --to 255.255.255.255:5064
+iptables-save > /etc/iptables/rules.v4
+
+usermod -a lnls-bpm -G dialout
+useradd iocs -m --no-user-group --shell /usr/sbin/nologin --home-dir /opt/container-iocs
```

Comment:
I think the idea would be to remove the user creation from the deploy script, since we should be able to trust this initial configuration was done correctly.

Reply:
Alright. Updated this in https://github.com/lnls-dig/bpm-app/pull/31.